src/lib/Parser.hs

```diff
@@ -336,20 +336,23 @@ decl = do
 interfaceInstance :: Parser UDecl
 interfaceInstance = do
   keyWord InstanceKW
-  (lhs, pos) <- withPos simpleLet
+  (p, pos) <- withPos $ letPat <|> parens pat
```

Comment:
Should this really be a `letPat` and not just a name? We don't want to allow patterns in there I think

Reply:
`letPat` is actually what we want, but `parens pat` should indeed be removed, thanks!